hugr-core/src/hugr/serialize/test.rs

```diff
@@ -114,35 +145,22 @@ fn empty_hugr_serialize() {
     check_hugr_roundtrip(&Hugr::default(), true);
 }
 
-/// Serialize and deserialize a value, optionally validating against a schema.
-pub fn ser_serialize_check_schema<T: Serialize + serde::de::DeserializeOwned>(
-    g: &T,
-    schema: Option<&JSONSchema>,
-) -> serde_json::Value {
-    let s = serde_json::to_string(g).unwrap();
-    let val: serde_json::Value = serde_json::from_str(&s).unwrap();
-
-    if let Some(schema) = schema {
-        let validate = schema.validate(&val);
-
-        if let Err(errors) = validate {
-            // errors don't necessarily implement Debug
-            for error in errors {
-                println!("Validation error: {}", error);
-                println!("Instance path: {}", error.instance_path);
-            }
-            panic!("Serialization test failed.");
-        }
-    }
-    val
+fn ser_deserialize_check_schema<T: serde::de::DeserializeOwned>(
+    val: serde_json::Value,
+    schemas: impl IntoIterator<Item = &'static NamedSchema>,
+) -> T {
+    NamedSchema::check_schemas(&val, schemas);
+    serde_json::from_value(val).unwrap()
 }
 
-/// Serialize and deserialize a HUGR, and check that the result is the same as the original.
-/// Checks the serialized json against the in-tree schema.
-///
-/// Returns the deserialized HUGR.
-pub fn check_hugr_schema_roundtrip(hugr: &Hugr) -> Hugr {
-    check_hugr_roundtrip(hugr, true)
+/// Serialize and deserialize a value, optionally validating against a schema.
```

Comment:
is this actually optional in this function?

Reply:
Good point, no it isn't.